sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2905,6 +2905,23 @@ object GpuOverrides extends Logging {
           )
         }
       }),
+    expr[ArraysZip](
+      "Returns a merged array of structs in which the N-th struct contains" +
+        " all N-th values of input arrays.",
+      ExprChecks.projectOnly(TypeSig.ARRAY.nested(
+        TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.NULL +
+          TypeSig.ARRAY + TypeSig.STRUCT + TypeSig.MAP),
+        TypeSig.ARRAY.nested(TypeSig.all),
+        repeatingParamCheck = Some(RepeatingParamCheck("children",
+          TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.NULL +
+            TypeSig.ARRAY + TypeSig.STRUCT + TypeSig.MAP),
+          TypeSig.ARRAY.nested(TypeSig.all)))),
+      (in, conf, p, r) => new ExprMeta[ArraysZip](in, conf, p, r) {
+        override def convertToGpu(): GpuExpression = {
+          GpuArraysZip(childExprs.map(_.convertToGpu()))
```

Comment:
https://issues.apache.org/jira/browse/SPARK-35876 updated the signature of Arrays.zip to avoid some issues in the optimizer. For the versions that we currently support it went into 3.2.0 and 3.1.3. It would be good to document this here, and have a follow on issue to pay off the tech debt once we no longer support 3.1.2 or below.

Reply:
Filed an issue #5327, and dropped the link here as a notice